workflows: various fixes & consistency passes

nbusseneau opened the pull request:
Please see individual commits :)

This supersedes the original #16595.

nbusseneau committed:
workflows: remove Azure SP from cilium install args

Following modifications to auto-generated Azure Service Principal in the
CLI, we can stop passing the main Azure SP to `cilium install`:
https://github.com/cilium/cilium-cli/pull/308

The auto-generated Service Principal is more secure as it is unique and
only has access to the cluster used for a specific workflow run.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

nbusseneau committed:
workflows: replace port-forwards with dedicated hubble command

Also refactor Cilium cleanup for a cleaner `pkill` pass.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

nbusseneau committed:
workflows: remove unnecessary relay deployment in 2nd cluster

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

nbusseneau committed:
workflows: standardize/comment all post-test and cleanup operations

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

nbusseneau committed:
workflows: remove connectivity test pods restart workaround

The flag `--force-deploy` does the same :)

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

nbusseneau committed:
workflows: consistently disabling flow validation

In #16388, we disabled flow validation as it was unstable in the CLI.
In #16617, we partially moved to `warning` for testing purposes.

This reverts back to `disabled` for all connectivity tests in order to
be consistent. Also fix missing disabled flow validation in EKS, as it
seems like we missed that one in #16388.

We can always enable flow validation again later once we are confident
flow validation works better in the CLI.

Fixes 49ce8e9953b877e6a9b2b1c71df1b767837cb690
Fixes 21c9b41c5ab5f580fa94591de9427f3a0965be35

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

nbusseneau committed:
workflows: properly setup multicluster cross-cluster traffic

- Both clusters created as VPC-native.
- Use common native routing CIDR.
- Cluster 2 inherits CA from cluster 1.
- Add dedicated firewall rule.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

nbusseneau committed:
workflows: fix tests ignore for multicluster testing

We changed that in v0.8 of the CLI: https://github.com/cilium/cilium-cli/pull/215

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>

pchaigno commented on .github/workflows/conformance-aks.yaml:
```diff
@@ -246,7 +246,7 @@ jobs:
       - name: Port forward Relay
         run: |
-          kubectl port-forward -n kube-system deployment/hubble-relay 4245:4245&
+          cilium hubble port-forward&
```
Maybe worth pulling in https://github.com/cilium/cilium-cli/pull/390 at the same time?

michi-covalent replied:
looks like it's blocked by this flake: https://github.com/cilium/cilium-cli/issues/399 😢

pchaigno replied:
Yes, but we don't need to wait for that commit to be merged in cilium-cli to pull it in cilium/cilium :-)

nbusseneau replied:
Thanks, added in 086ff29af18d85c02848a2dced47683cec3cca97.

nbusseneau committed:
workflows: check status of Relay port-forward

Since we start a background process for port-forwarding Relay, we must
check if it actually succeeded or failed before allowing workflows to
continue.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
